glean-core/src/traits/jwe.rs

```diff
@@ -0,0 +1,41 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+/// A description for the `JweMetric` type.
+///
+/// When changing this trait, make sure all the operations are
+/// implemented in the related type in `../metrics/`.
+pub trait Jwe {
+    /// Sets to the specified JWE value.
+    ///
+    /// # Arguments
+    ///
+    /// * `value` - the [`compact representation`](https://tools.ietf.org/html/rfc7516#appendix-A.2.7) of a JWE value.
+    fn set_with_compact_representation<S: Into<String>>(&self, value: S);
+
+    /// Builds a JWE value from its elements and set to it.
+    ///
+    /// # Arguments
+    ///
+    /// * `header` - the JWE Protected Header element.
+    /// * `key` - the JWE Encrypted Key element.
+    /// * `init_vector` - the JWE Initialization Vector element.
+    /// * `cipher_text` - the JWE Ciphertext element.
+    /// * `auth_tag` - the JWE Authentication Tag element.
+    fn set<S: Into<String>>(&self, header: S, key: S, init_vector: S, cipher_text: S, auth_tag: S);
```

Comment:
Uh oh, this actually means all parameters need to be of the same type.
Not a problem for our current use, but we should consider changing that.
(actually a nicer Rust-y API would probably to take a `JweToken` struct or something, I'll file a bug here as well)

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1672183